py2pack/__init__.py

```diff
@@ -302,11 +307,13 @@ def main():
     parser_fetch = subparsers.add_parser('fetch', help='download package source tarball from PyPI')
     parser_fetch.add_argument('name', help='package name')
     parser_fetch.add_argument('version', nargs='?', help='package version (optional)')
+    parser_fetch.add_argument('--source-url', nargs='?', default='', help='source url')
```

Comment:
Not sure that you need `nargs` for this parameter

Reply:
oops, you're right. Fixed

Reply:
In fact, I forgot that removing nargs removed the source_url attribute with default value from  args, which makes it fail later in newest_download_url (where it's assumed to exist). I'm not sure about the preferred coding style, so I'll ask. Which of the following solutions is preferred? (have a look at travis below to see the problem).

1. Leave ```nargs='?'``` which makes it always include a source_url attribute in args (with default value if the user didn't specify the argument)
2. Use ```getattr(args, 'source_url', None)``` to get a default value when args doesn't contain a source_url attribute.
3. Use ```try: ... except AttributeError: pass```  each time it's used to ignore the error if the attribute is not available.

Reply:
I am not sure that this is correct. I can be wrong but the `default=''` makes args populated with `source_url` without the `nargs` in any case. The test error is because the test is injecting the `Args` object that is a mockup.